numpy/core/code_generators/numpy_api.py

```diff
@@ -223,131 +223,132 @@
     'PyArray_BoolConverter':                (179,),
     'PyArray_ByteorderConverter':           (180,),
     'PyArray_OrderConverter':               (181,),
-    'PyArray_EquivTypes':                   (182,),
-    'PyArray_Zeros':                        (183, StealRef(3)),
-    'PyArray_Empty':                        (184, StealRef(3)),
-    'PyArray_Where':                        (185,),
-    'PyArray_Arange':                       (186,),
-    'PyArray_ArangeObj':                    (187,),
-    'PyArray_SortkindConverter':            (188,),
-    'PyArray_LexSort':                      (189,),
-    'PyArray_Round':                        (190,),
-    'PyArray_EquivTypenums':                (191,),
-    'PyArray_RegisterDataType':             (192,),
-    'PyArray_RegisterCastFunc':             (193,),
-    'PyArray_RegisterCanCast':              (194,),
-    'PyArray_InitArrFuncs':                 (195,),
-    'PyArray_IntTupleFromIntp':             (196,),
-    'PyArray_TypeNumFromName':              (197,),
-    'PyArray_ClipmodeConverter':            (198,),
-    'PyArray_OutputConverter':              (199,),
-    'PyArray_BroadcastToShape':             (200,),
-    '_PyArray_SigintHandler':               (201,),
-    '_PyArray_GetSigintBuf':                (202,),
-    'PyArray_DescrAlignConverter':          (203,),
-    'PyArray_DescrAlignConverter2':         (204,),
-    'PyArray_SearchsideConverter':          (205,),
-    'PyArray_CheckAxis':                    (206,),
-    'PyArray_OverflowMultiplyList':         (207,),
-    'PyArray_CompareString':                (208,),
-    'PyArray_MultiIterFromObjects':         (209,),
-    'PyArray_GetEndianness':                (210,),
-    'PyArray_GetNDArrayCFeatureVersion':    (211,),
-    'PyArray_Correlate2':                   (212,),
-    'PyArray_NeighborhoodIterNew':          (213,),
+    'PyArray_SizeConverter':                (182,),
```

Comment:
I'll be totally honest, I have no idea what I did here or why. The error messages made it pretty clear that the indices need to be unique, but I have no idea why or what they do. Should I have just set this one to 304 (the next available number)? If so, why aren't they all just being generated?

Reply:
These numbers are not allowed to change, because that will break ABI compatibility for code compiled against previous versions of numpy - so your new number must go at the end,

Reply:
They are indices in a table of function pointers, which are used to export the functions to other python extension modules.